tests/v2/validation/networking/connectivity/port_test.go

```diff
@@ -302,13 +510,90 @@ func (p *PortTestSuite) TestLoadBalance() {
 			NodePort:   int32(nodePort),
 		},
 	}
-	lbService := services.NewServiceTemplate(serviceName, p.namespace.Name, corev1.ServiceTypeLoadBalancer, ports, daemonsetTemplate.Spec.Selector.MatchLabels)
-	serviceResp, err := services.CreateService(steveClient, lbService)
+
+	clusterIPService := services.NewServiceTemplate(serviceName, namespace.Name, corev1.ServiceTypeLoadBalancer, ports, deploymentTemplate.Spec.Template.Labels)
+	serviceResp, err := services.CreateService(steveClient, clusterIPService)
 	require.NoError(p.T(), err)
 
 	err = services.VerifyService(steveClient, serviceResp)
 	require.NoError(p.T(), err)
 
+	log.Info("Scaling up deployment")
+	replicas = int32(3)
+	deploymentTemplate.Spec.Replicas = &replicas
+	deploymentTemplate, err = deployment.UpdateDeployment(p.client, p.cluster.ID, namespace.Name, deploymentTemplate, true)
+	require.NoError(p.T(), err)
+	p.validateWorkload(deploymentTemplate, containerImage, 3, namespace.Name)
+	p.validateLoadBalance(steveClient, nodePort, deploymentName)
+
+	log.Info("Scaling down deployment")
+	replicas = int32(2)
+	deploymentTemplate.Spec.Replicas = &replicas
+	deploymentTemplate, err = deployment.UpdateDeployment(p.client, p.cluster.ID, namespace.Name, deploymentTemplate, true)
+	require.NoError(p.T(), err)
+	p.validateWorkload(deploymentTemplate, containerImage, 2, namespace.Name)
+	p.validateLoadBalance(steveClient, nodePort, deploymentName)
+
+	log.Info("Upgrating deployment")
+	for _, c := range deploymentTemplate.Spec.Template.Spec.Containers {
+		c.Name = namegen.AppendRandomString("test-upgrade")
+	}
+
+	log.Info("Updating deployment replicas")
+	deploymentTemplate, err = deployment.UpdateDeployment(p.client, p.cluster.ID, namespace.Name, deploymentTemplate, true)
+	require.NoError(p.T(), err)
+	p.validateWorkload(deploymentTemplate, containerImage, 2, namespace.Name)
+	p.validateLoadBalance(steveClient, nodePort, deploymentName)
+}
+
+// This must be a valid port number, 10250 < hostPort < 65536
+// The range 1-10250 are reserved to Rancher
+// Using a random port to avoid 'port in use' failures and allow the test to be rerun
+func getHostPort() int {
+	return rand.IntN(55283) + 10251
```

Comment:
This line is reused in multiple functions, so it would make sense to centralize this logic in a single function

Reply:
I see that this line is reused in multiple functions, but since it’s a simple one liner, keeping it inline in each test might help with readability without adding an extra function. Adding a function could be overkill here, but if you feel strongly about it, we can have it as is. Just a minor suggestion